staging/src/k8s.io/client-go/pkg/controllers/lease/controller.go

```diff
@@ -170,10 +182,14 @@ func (c *controller) ensureLease() (*coordinationv1.Lease, bool, error) {
 // call this once you're sure the lease has been created
 func (c *controller) retryUpdateLease(base *coordinationv1.Lease) error {
 	for i := 0; i < maxUpdateRetries; i++ {
-		lease, err := c.leaseClient.Update(context.TODO(), c.newLease(base), metav1.UpdateOptions{})
-		if err == nil {
-			c.latestLease = lease
-			return nil
+		leaseToUpdate, blockUpdate, err := c.newLease(base)
+		if !blockUpdate {
```

Comment:
Can you use the opposite condition? i.e.,

```golang
if blockUpdate {
  klog.Error(...)
  continue
}
lease, err = c.leaseClient.Update(context.TODO(), leaseToUpdate, metav1.UpdateOptions{})
if err == nil {
  c.latestLease = lease
  return nil
}
klog.Error(...)
```


Also move the onRepeatedHeartbeatFailure check to the beginning of the for loop to keep the semantics.